core/src/main/java/dk/alexandra/fresco/lib/real/fixed/AdvancedFixedNumeric.java

```diff
@@ -0,0 +1,49 @@
+package dk.alexandra.fresco.lib.real.fixed;
+
+import dk.alexandra.fresco.framework.DRes;
+import dk.alexandra.fresco.framework.builder.numeric.AdvancedNumeric.RandomAdditiveMask;
+import dk.alexandra.fresco.framework.builder.numeric.ProtocolBuilderNumeric;
+import dk.alexandra.fresco.framework.value.SInt;
+import dk.alexandra.fresco.lib.real.DefaultAdvancedRealNumeric;
+import dk.alexandra.fresco.lib.real.SReal;
+import java.math.BigDecimal;
+
+public class AdvancedFixedNumeric extends DefaultAdvancedRealNumeric {
+
+  private static final int DEFAULT_RANDOM_BITS = 64;
```

Comment:
While this is in some sense better than before it does not solve the underlying problem. See comments on the `AdvancedRealNumeric` interface.